compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -12759,6 +12960,12 @@ TR::Node *ificmpeqSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier
 
 
    addressCompareConversion(node, s);
+
+   if (s->getLastRun() && !s->comp()->getOption(TR_disableIntegerCompareSimplification))
```

Comment:
I am not a fan of this control logic being sprayed everywhere - there should be an early return in removeArithmeticsUnderIntegralCompare so that changes to this logic are easily made.